recipes/lz4/all/conanfile.py

```diff
@@ -12,106 +8,67 @@ class LZ4Conan(ConanFile):
-               self.settings.compiler == "gcc" and \
-              os.name == "nt"
-
     def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
         del self.settings.compiler.libcxx
         del self.settings.compiler.stdcpp
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
-    def build_requirements(self):
-        if self._is_mingw_windows:
-            self.build_requires("msys2/20161025")
-
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_folder = "{0}-{1}".format(self.name, self.version)
         os.rename(extracted_folder, self._source_subfolder)
 
-    def _build_make(self):
-        prefix = self.package_folder
-        prefix = tools.unix_path(prefix) if self.settings.os == "Windows" else prefix
-        with tools.chdir(self._source_subfolder):
-            env_build = AutoToolsBuildEnvironment(self)
-            if self.options.shared:
-                args = ["BUILD_SHARED=yes", "BUILD_STATIC=no"]
-            else:
-                args = ["BUILD_SHARED=no", "BUILD_STATIC=yes"]
-            env_build.make(args=args)
-            args.extend(["PREFIX=%s" % prefix, "install"])
-            env_build.make(args=args)
-
-            tools.rmdir(os.path.join(prefix, "share"))
-            tools.rmdir(os.path.join(prefix, "lib", "pkgconfig"))
-            tools.rmdir(os.path.join(prefix, "bin"))
-
-            if self.settings.os == 'Macos' and self.options.shared:
-                lib_dir = os.path.join(prefix, 'lib')
-                old = '/usr/local/lib/liblz4.1.dylib'
-                new = 'liblz4.1.dylib'
-                for lib in os.listdir(lib_dir):
-                    if lib.endswith('.dylib'):
-                        self.run('install_name_tool -change %s %s %s' % (old, new, os.path.join(lib_dir, lib)))
-
-    def _build_vs(self):
-        shutil.copy(os.path.join(self._source_subfolder, "lib", "lz4.h"),
-                    os.path.join(self._source_subfolder, "visual", "VS2010", "liblz4-dll", "lz4.h"))
-        # Unable to load plug-in localespc.dll
-        for project in ["lz4", "liblz4", "liblz4-dll"]:
-            project_name = os.path.join(self._source_subfolder, "visual", "VS2010", project, "%s.vcxproj" % project)
-            tools.replace_in_file(project_name, "<RunCodeAnalysis>true</RunCodeAnalysis>", "")
-            tools.replace_in_file(project_name, "<TreatWarningAsError>true</TreatWarningAsError>", "")
-        with tools.chdir(os.path.join(self._source_subfolder, 'visual', 'VS2010')):
-            target = 'liblz4-dll' if self.options.shared else 'liblz4'
-
-            msbuild = MSBuild(self)
-            msbuild.build(project_file="lz4.sln", targets=[target], platforms={'x86': 'Win32'})
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["LZ4_BUNDLED_MODE"] = False
+        if "fPIC" in self.options:
```

Comment:
You can use `self.options.get_safe("fPIC")` but I'm okay with your implementation.